receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -51,7 +51,7 @@ type metricGroup struct {
 	ls           labels.Labels
 	count        float64
 	hasCount     bool
-	sum          float64
+	sum          *float64
 	hasSum       bool
```

Comment:
There is a "hasSum" why pointer?

Reply:
hasSum is never used anywhere. :).  Probably something we should rip out...  Or do you think we should use a separate bool instead of a pointer to the float?

Reply:
I think separate bool is better, since using here a pointer forces that float to be heap allocated.

Reply:
done